### PR TITLE
:sparkles: Validators now execute asnychronously

### PR DESCRIPTION
**Pull request by fabianvf**
- Also adds error handling around task management to avoid crashes

### CODE REVIEW

File: kai/reactive_codeplanner/task_manager/task_manager.py [hunk 1]
```diff
@@ -93,21 +99,53 @@ def run_validators(self) -> list[Task]:
         logger.info("Running validators.")
         validation_tasks: list[Task] = []
 
-        for validator in self.validators:
+        def run_validator(validator):
```
Comment (shawn-hurley) [inline kai/reactive_codeplanner/task_manager/task_manager.py:102]: whats the return of this?

File: kai/reactive_codeplanner/task_manager/task_manager.py [hunk 2]
```diff
@@ -93,21 +99,53 @@ def run_validators(self) -> list[Task]:
...
+                return validator, None
+
+        with ThreadPoolExecutor() as executor:
+            # Submit all validators to the executor
+            future_to_validator = {
+                executor.submit(run_validator, v): v for v in self.validators
+            }
+
+            # Process results as they complete
+            for future in as_completed(future_to_validator):
```
Comment (shawn-hurley) [inline kai/reactive_codeplanner/task_manager/task_manager.py:122]: Should we have a timeout variable?

**Review by shawn-hurley — APPROVED** [review]
LGTM to me, just a few questions

**Comment by fabianvf** [comment]
> I just gave this a test, and seeing that an error in the validator causes everything to block
> 
> ```
> WARNING - 2024-11-11 11:57:53,005 - kai.kai.reactive_codeplanner.task_manager.task_manager - MainThread - [task_manager.py:95 - supply_result()] - Encountered errors: [ApiResponseException('Failed to handle request to https://bam-api.res.ibm.com/v2/text/chat_stream?version=2024-01-10.\n{\n  "error": "Bad Request",\n  "extensions": {\n    "code": "INVALID_INPUT",\n    "state": null\n  },\n  "message": "Messages must be in order: [s]/u/a/u/a/u",\n  "status_code": 400\n}')]
> ```

I think this error isn't coming from a validator, but from an agent. The validators don't use LLMs AFAIK

File: kai/reactive_codeplanner/task_manager/task_manager.py (hunk 2 above)
Comment (fabianvf) [reply kai/reactive_codeplanner/task_manager/task_manager.py:122]: Probably eventually but I have no idea what a good timeout would be. For the analyzer for example it'll scale based on codebase, on # of rules, etcetc

**Pull request merged**